source/whats-new.txt

```diff
@@ -26,6 +27,53 @@ Learn what's new in:
 * :ref:`Version 3.7 <version-3.7>`
 * :ref:`Version 3.6 <version-3.6>`
 
+.. _version-4.11:
+
+What's New in 4.11
+------------------
+
+The 4.11 {+driver-short+} release includes **recursive schema support**,
+meaning that you can use recursive types as the schema in the
+``Collection`` class. The driver also provides type safety for dot
+notation queries up to a depth of nine in this release. Beyond a depth
+of nine, TypeScript compiles your code, but property types fall back
+to ``any`` type at that level. The recursive type depth limit is a
+current limitation in TypeScript.
+
+Recursive Schema Type Checking Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose we have a collection of type ``Collection<CircularSchema>``:
+
+.. code-block:: js
+   :copyable: false
+
+   interface CircularSchema {
+       name: string;
+       ref: CircularSchema;
+   }
+
+TypeScript enforces type checking below a depth of nine. The following
+code causes a TypeScript compilation error because the ``name`` property
+value must be a ``string`` type:
+
+.. code-block:: js
+   :copyable: false
+
+   collection.findOne({ 'ref.ref.ref.name': 25 })
+
+At a depth greater than nine, TypeScript compiles your code but no
+longer type checks it. The following code references a property at a
+depth of 11 and does not cause a compilation error:
```

Comment:
Suggestion: I think an explanation of what doesn't cause the compilation error could be helpful rather than inferring from the code example. E.g. "assigns an integer to a string property, but does not cause a compilation error because the property is at a depth of 11"